src/Validation/URLValidationCron.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * WP cron process to validate URLs in the background.
+ *
+ * @package AMP
+ * @since 2.1
+ */
+
+namespace AmpProject\AmpWP\Validation;
+
+use AmpProject\AmpWP\BackgroundTask\BackgroundTaskDeactivator;
+use AmpProject\AmpWP\BackgroundTask\RecurringBackgroundTask;
+
+/**
+ * URLValidationCron class.
+ *
+ * @since 2.1
+ *
+ * @internal
+ */
+final class URLValidationCron extends RecurringBackgroundTask {
+
+	/**
+	 * ScannableURLProvider instance.
+	 *
+	 * @var ScannableURLProvider
+	 */
+	private $scannable_url_provider;
+
+	/**
+	 * URLValidationProvider instance.
+	 *
+	 * @var URLValidationProvider
+	 */
+	private $url_validation_provider;
+
+	/**
+	 * The cron action name.
+	 *
+	 * @var string
+	 */
+	const BACKGROUND_TASK_NAME = 'amp_validate_urls';
+
+	/**
+	 * The length of time, in seconds, to sleep between each URL validation.
+	 *
+	 * @var int
+	 */
+	const DEFAULT_SLEEP_TIME = 1;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param BackgroundTaskDeactivator $background_task_deactivator Service that deactivates background events.
+	 * @param ScannableURLProvider      $scannable_url_provider ScannableURLProvider instance.
+	 * @param URLValidationProvider     $url_validation_provider URLValidationProvider instance.
+	 */
+	public function __construct( BackgroundTaskDeactivator $background_task_deactivator, ScannableURLProvider $scannable_url_provider, URLValidationProvider $url_validation_provider ) {
+		parent::__construct( $background_task_deactivator );
+
+		$this->scannable_url_provider  = $scannable_url_provider;
+		$this->url_validation_provider = $url_validation_provider;
+	}
+
+	/**
+	 * Callback for the cron action.
+	 *
+	 * @param mixed[] ...$args Unused callback arguments.
+	 */
+	public function process( ...$args ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$urls       = $this->scannable_url_provider->get_urls();
+		$sleep_time = $this->get_sleep_time();
+
+		foreach ( $urls as $url ) {
+			$this->url_validation_provider->get_url_validation( $url['url'], $url['type'], true );
+			if ( $sleep_time ) {
+				sleep( $sleep_time );
+			}
+		}
+	}
```

Comment:
We can address this in a subsequent PR, but I have an idea for how this could be modified to not make use of `sleep()`. Instead of there being one single event that scans all of the URLs, we can instead register a separate event for each for each URL, and offset the `time()` for each event by 10 minute intervals (for example). In this way only one URL would be validated during any given cron request, and there would be no need for `sleep()`.

Reply:
See #5750.